pkg/provision/cleanup.go

```diff
@@ -67,50 +75,81 @@ func cleanupResources(msgBrokerCon string) error {
 
 		for _, subvol := range subvols {
 			log.Info().Msgf("checking subvol %s", subvol.Path)
-			_, ok := qgroups[fmt.Sprintf("0/%d", subvol.ID)]
+			// Don't delete zos-cache!
+			if subvol.Path == "zos-cache" {
+				continue
+			}
+
+			qgroup, ok := qgroups[fmt.Sprintf("0/%d", subvol.ID)]
 			if !ok {
 				log.Info().Msgf("skipping volume '%s' has no assigned quota", subvol.Path)
 				continue
 			}
 
-			// now, is this subvol in one of the toDelete ?
+			// Now, is this subvol in one of the toDelete ?
 			if _, ok := toDelete[filepath.Base(subvol.Path)]; ok {
 				// delete the subvolume
-				log.Info().Msgf("deleting subvolume %s", subvol.Path)
-				// if err := utils.SubvolumeRemove(ctx, filepath.Join(path, subvol.Path)); err != nil {
-				// 	log.Err(err).Msgf("failed to delete subvol '%s'", subvol.Path)
-				// }
-				// if err := utils.QGroupDestroy(ctx, qgroup.ID, path); err != nil {
-				// 	log.Err(err).Msgf("failed to delete qgroup: '%s'", qgroup.ID)
-				// }
+				delete := checkReservationToDelete(subvol.Path, client)
+				if delete {
```

Comment:
We need the following:

1: check if the volume need to be kept, if yes, continue
2: check if the volume needs to be deleted, if yes, check the explorer
3: check if the volume is any other volume that is not present in the toSave and toDelete map. Since we create and fill these maps based on containers, but I have seen some cases where there would be other volumes as well which are not linked to containers

Reply:
But shouldn't you also list all the subvolumes present on the node ? How can you find volume not attached with container otherwise ?